pkg/utils/node/predicates.go

```diff
@@ -54,3 +55,9 @@ func IsUnderutilized(node *v1.Node, pods []*v1.Pod) bool {
 	}
 	return true
 }
+
+func IsKarpenterProvisioned(node *v1.Node) bool {
+	_, okName := node.Labels[v1alpha1.ProvisionerNameLabelKey]
+	_, okNamespace := node.Labels[v1alpha1.ProvisionerNamespaceLabelKey]
+	return okName && okNamespace
```

Comment:
Could it ever happen that one of these is set but the other is not? If so, how would that happen?

Reply:
As I understand it, this could happen if it's removed manually, or if someone who wants a node to become controlled by a Karpenter controller who only patched in one of the labels. 

In the case that we have only one of the labels, we won't be blocked from doing work, and we can still properly terminate the instance. I can see doing `&&` or `||` here in both cases. `&&` enforces that each node must have the proper labels, so that our other controller can discover the node and do work. `||` allows nodes that aren't perfect to be included, making it easier for work to happen.